feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt

```diff
@@ -7,35 +7,68 @@ import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringArrayResource
 import androidx.compose.ui.res.stringResource
-import androidx.core.text.isDigitsOnly
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.iteneum.RepairRequest
+import com.iteneum.UIEventRepair
 import com.iteneum.designsystem.components.*
 import com.iteneum.designsystem.theme.LPTypography
 import com.iteneum.designsystem.theme.LeasePertTheme
 import com.iteneum.repair.R
+import com.iteneum.repair.data.RepairViewModel
 
 /**
  * Function that creates [RepairView] compose. for user's service request.
  * User will require to fill all the correspondent fields on the screen.
  *
+ * Function [RepairView] has
+ * @param navigateToApartment - to get the function to navigate ApartmentView when clicking button
+ * @param repairViewModel - to get the Repair View Model access
+ *
+ * Function [RepairContainer] has
+ * @param formularyValues - to pass the content of the RepairRequest
+ * @param formularyData - to handle the UI events of data when text is modified
+ * @param validatePhone - to validate is thw phone contains 10 digits and only numbers
+ * @param onClickButton - to handle the action when button is clicked
+ * @param navigateToApartment - passed from RepairView to handle navigation
+ *
  * @author Jose Miguel Garcia Reyes
  */
 
+@Composable
+fun RepairView(
+    navigateToApartment: () -> Unit,
+    repairViewModel: RepairViewModel = hiltViewModel()
+) {
+    LaunchedEffect(true) {
+        repairViewModel.getInformation()
+    }
+    RepairContainer(
+        formularyValues = repairViewModel.repairRequest,
+        formularyData = {
+            repairViewModel.setValues(it)
+        },
+        validatePhone = { repairViewModel.validatePhone(phone = it) },
+        onClickButton = { repairViewModel.onClickSendButton() },
+        navigateToApartment = navigateToApartment
+    )
+}
+
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun RepairView() {
+fun RepairContainer(
+    formularyValues: RepairRequest,
+    formularyData: (UIEventRepair) -> Unit,
+    validatePhone: (it: String) -> Boolean,
+    onClickButton: () -> Unit,
+    navigateToApartment: () -> Unit
+) {
     val sizes = LeasePertTheme.sizes
     val optionsPermissionRadioButtons = stringArrayResource(id = R.array.options_radio_button)
     val optionsPetInUnit = stringArrayResource(id = R.array.options_pet_in_unit)
     val optionsCategory = stringArrayResource(id = R.array.options_category)
     var isNotValidPhone by remember {
         mutableStateOf(false)
     }
-    var optionSelectedRadioButtons by remember {
-        mutableStateOf(optionsPermissionRadioButtons[0])
-    }
-    var contactPhoneNumberValue by remember {
-        mutableStateOf("")
-    }
     LeasePertTheme {
```

Comment:
I think **LeasePertTheme** is not necesary, try delete it and it should works well

Reply:
Ok, removed it!